apps/docs/content/guides/ai/hybrid-search.mdx

```diff
@@ -0,0 +1,228 @@
+
+## Hybrid search in Postgres
+
+Let's implement hybrid search in Postgres using `tsvector` (keyword search) and `pgvector` (semantic search).
+
+First we'll create a `documents` table to store the documents that we will search over. This is just an example - adjust this to match the structure of your application.
+
+```sql
+create table documents (
+  id bigint primary key generated always as identity,
+  content text,
+  fts tsvector generated always as (to_tsvector('english', content)) stored,
+  embedding vector(512)
+);
+```
+
+The table contains 4 columns:
+
+- `id` is an auto-generated unique ID for the record. We'll use this later to match records when performing RRF.
+- `content` contains the actual text we will be searching over.
+- `fts` is an auto-generated `tsvector` column that is generated using the text in `content`. We will use this for [full text search](/docs/guides/database/full-text-search) (search by keyword).
+- `embedding` is a [vector column](/docs/guides/ai/vector-columns) that stores the vector generated from our embedding model. We will use this for [semantic search](/docs/guides/ai/concepts#what-are-embeddings) (search by meaning). We chose 512 dimensions for this example, but adjust this to match the size of the embedding vectors generated from your preferred model.
+
+Next we'll create indexes on the `fts` and `embedding` columns so that their individual queries will remain fast at scale:
+
+```sql
+-- Create an index for the full-text search
+create index on documents using gin(fts);
+
+-- Create an index for the semantic vector search
+create index on documents using hnsw (embedding vector_ip_ops);
+```
+
+For full text search we use a [generalized inverted (GIN) index](https://www.postgresql.org/docs/current/gin-intro.html) which is designed for handling composite values like those stored in a `tsvector`.
+
+For semantic vector search we use an [HNSW index](/docs/guides/ai/vector-indexes/hnsw-indexes), which is a high performing approximate nearest neighbor (ANN) search algorithm. Note that we are using the `vector_ip_ops` (inner product) operator with this index because we plan on using the inner product (`<#>`) operator later in our query. If you plan to use a different operator like cosine distance (`<=>`), be sure to update the index accordingly. For more information, see [distance operators](/docs/guides/ai/vector-indexes#distance-operators).
+
+Finally we'll create our `hybrid_search` function:
+
+```sql
+create or replace function hybrid_search(
+  query_text text,
+  query_embedding vector(512),
+  match_count int,
+  full_text_weight float = 1,
+	semantic_weight float = 1,
+  rrf_k int = 50
+)
+returns setof documents
+language sql
+as $$
+with full_text as (
+  select
+    id,
+    row_number() over() as rank_ix
+  from
+    documents
+  where
+    fts @@ websearch_to_tsquery(query_text)
```

Comment:
Changed `query_text` to a `text` type so that it can easily be called from the Supabase client libs (not possible to pass in a `tsvector` directly). The consequence is we have to choose a wrapping function - I'm choosing `websearch_to_tsquery(query_text)` here as the safest default for web apps.

@olirice Any performance concerns calling `websearch_to_tsquery` here and below in the `order by` (ie. is it cached)?

Reply:
What you have currently is best

Here are the 3 options I tried
<img width="755" alt="Screenshot 2024-03-12 at 11 02 49 AM" src="https://github.com/supabase/supabase/assets/12958657/e4cec7b7-2931-4907-9395-fae28e9eebf1">
<img width="755" alt="Screenshot 2024-03-12 at 11 02 09 AM" src="https://github.com/supabase/supabase/assets/12958657/da9069e0-c44a-4ae2-a5cb-e1f9066d6fb1">
<img width="774" alt="Screenshot 2024-03-12 at 11 01 15 AM" src="https://github.com/supabase/supabase/assets/12958657/32c666d3-0c6a-4a34-a0f6-67ef193dfe9c">